cwl_utils/sandboxjs.py

```diff
@@ -308,6 +308,8 @@ def new_js_proc(
         if nodejs is None or nodejs is not None and required_node_version is False:
             try:
                 nodeimg = "docker.io/node:slim"
+                singularityimgs = [0]
```

Comment:
Don't you want this to be the empty list, `[]` instead of `[0]` ?
-- later edit, after disecting the if statement later on I understand you make singularityimgs non empty so that the condition to test whether to pull remains valid when the container engine is not singularity. This is a bit hard to grasp; I suggest changing the `if` statement as suggested later, and removing this statement here.